reporters/reporter-default/src/main/java/org/radargun/reporting/html/HtmlDocument.java

```diff
@@ -29,61 +28,29 @@ public HtmlDocument(String directory, String fileName, String title) {
       this.title = title;
    }
 
-   /**
-    * Open the document file and write common headers.
-    * @throws IOException
-    */
-   public void open() throws IOException {
-      File dir = new File(directory);
-      if (dir.exists() && !dir.isDirectory()) {
-         throw new IllegalArgumentException(dir.getAbsolutePath() + " is not a directory");
-      } else if (!dir.exists()) {
-         dir.mkdirs();
-      }
-      writer = new PrintWriter(directory + File.separator + fileName);
-      write("<HTML><HEAD><TITLE>");
-      write(title);
-      write("</TITLE><STYLE>\n");
-      writeStyle();
-      write("</STYLE>\n<SCRIPT>\n");
-      writeScripts();
-      write("</SCRIPT></HEAD>\n<BODY>");
+   public String getTitle() {
+      return title;
    }
 
-   /**
-    * Write CSS styles.
-    */
-   protected void writeStyle() {
+   public String getFileName() {
+      return fileName;
    }
 
-   /**
-    * Write JavaScript scripts used in the document.
-    */
-   protected void writeScripts() {
+   public String getDirectory() {
+      return directory;
    }
 
-   /**
-    * Write footer and close the file.
-    */
-   public void close() {
-      writer.println("</BODY></HTML>");
-      writer.close();
+   public void createReportDirectory() {
+      File dir = new File(directory);
+      if (dir.exists() && !dir.isDirectory()) {
+         throw new IllegalArgumentException(dir.getAbsolutePath() +
+                 " is not a directory");
+      } else if (!dir.exists()) {
```

Comment:
dit.mkdirs() is enough and if returns true then all directiry tree created otherwise not created. Check here http://docs.oracle.com/javase/7/docs/api/java/io/File.html#mkdirs%28%29